### PR TITLE
OSDOCS-3097: Monitoring overview

### PR DESCRIPTION
**Pull request by Amrita42**
- Applies to 4.8+ versions
- [Jira](https://issues.redhat.com/browse/OSDOCS-3097)
- [Preview](https://deploy-preview-40169--osdocs.netlify.app/openshift-enterprise/latest/monitoring/overview-of-monitoring.html)
- [4.7 PR](https://github.com/openshift/openshift-docs/pull/40170) (cherry picked to 4.6)
- @codyhoag  Can you please peer-review ?
- @bburt-rh could you please review. Since Understanding the monitoring stack is an overview, it needs to include all the relevant sections in the book. Once you give me feedback, I will incorporate the same in the related 4.7 PR.

### CODE REVIEW

File: monitoring/index.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[id="monitoring-overview"]
+= Monitoring overview
+include::modules/common-attributes.adoc[]
+:context: monitoring-overview
+
+toc::[]
+
+The growth of containerization has not only brought many advantages to developers,DevOps, and IT teams but also challenges. The flexibility and scalability of containerized applications pose challenges since there is no direct relationship between an application and the server it runs on.
```
Comment (juzhao) [inline monitoring/index.adoc:8]: `developers, `, should add a space after `,`

File: monitoring/index.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+The monitoring stack is available by default, when {product-title} is installed. The cluster administrator has the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
+You can also xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations.
+You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to:
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
+* xref:../monitoring/reviewing-monitoring-dashboards.adoc#reviewing-monitoring-dashboards[Review monitoring dashboards]
+
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics,alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI.
```
Comment (juzhao) [inline monitoring/index.adoc:22]: `metrics, `,  should add a space after `,`

File: monitoring/index.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+As a cluster administrator you can determine why:
+
+* xref:../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues[User-metrics are not available]
```
Comment (juzhao) [inline monitoring/index.adoc:26]: It is better change the link to
`../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues#investigating-why-user-defined-metrics-are-unavailable_troubleshooting-monitoring-issues`

File: monitoring/index.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+* xref:../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues[Prometheus consumes a lot of disk space]
```
Comment (juzhao) [inline monitoring/index.adoc:27]: It is better change the link to
`../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues#determining-why-prometheus-is-consuming-disk-space_troubleshooting-monitoring-issues`

File: monitoring/index.adoc (hunk 3 above)
Comment (Amrita42) [reply monitoring/index.adoc:26]: Due to tooling limitations, we dont xref to modules

File: monitoring/index.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+include::modules/common-attributes.adoc[]
+:context: monitoring-overview
+
+toc::[]
+
+The growth of containerization has not only brought many advantages to developers, DevOps, and IT teams but also challenges. The flexibility and scalability of containerized applications pose challenges since there is no direct relationship between an application and the server it runs on.
+A cluster administrator can use the pre-installed, pre-configured and self updating {product-title} xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[monitoring stack] to monitor:
+
+* Default core platform components
+* (Optional) components for user-defined projects
```
Comment (codyhoag) [inline monitoring/index.adoc:12]: ISG is vague on this, but I personally prefer something like this:

`* Components for user-defined projects (Optional)`

If you choose to keep it as-is, I'd recommend capitalizing `components`

File: monitoring/index.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+The monitoring stack is available by default, when {product-title} is installed. The cluster administrator has the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
```
Comment (codyhoag) [inline monitoring/index.adoc:14]: I would remove the comma after `default`

File: monitoring/index.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+You can also xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations.
+You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to:
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
+* xref:../monitoring/reviewing-monitoring-dashboards.adoc#reviewing-monitoring-dashboards[Review monitoring dashboards]
+
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics, alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI.
```
Comment (codyhoag) [inline monitoring/index.adoc:22]: The first sentence here repeats the bullet list above. Is there a way this information can be combined?

File: monitoring/index.adoc [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+As a cluster administrator you can determine why:
```
Comment (codyhoag) [inline monitoring/index.adoc:24]: With our xref restrictions, perhaps we could link once to the troubleshooting guide and list the topics. Maybe something like this:

```
As a cluster administrator, you can xref:../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues[troubleshoot monitoring issues]. Common issues include:

* User-metrics are not available
* Prometheus consumes a lot of disk space
```

File: _topic_maps/_topic_map.yml [hunk 9]
```diff
@@ -1984,6 +1984,8 @@ Name: Monitoring
 Dir: monitoring
 Distros: openshift-enterprise,openshift-origin
 Topics:
+- Name: Monitoring Overview
```
Comment (codyhoag) [inline _topic_maps/_topic_map.yml:1987]:
```suggestion
- Name: Monitoring overview
```

**Review by codyhoag — COMMENTED** [review]
Just a few nits; looks good otherwise!

File: monitoring/index.adoc (hunk 5 above)
Comment (Amrita42) [reply monitoring/index.adoc:12]: I will go with your suggestions

**Review by ctauchen — COMMENTED** [review]
I've left a few line edits and suggestions in the comments below.

Forgive me for asking a very broad and speculative question at this late stage. Overview articles are difficult. After reading this and the next assembly in the book (Understanding monitoring), I'm left wondering exactly what this overview does differently from Understanding monitoring, which itself provides an in-depth overview of the monitoring stack.

If Understanding monitoring is actually an overview article (even if not in name), one approach here might be to add new information to Understanding, such as additional signposts and links to specific areas of the book?

File: monitoring/index.adoc [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics, alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI. You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to:
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
+* xref:../monitoring/reviewing-monitoring-dashboards.adoc#reviewing-monitoring-dashboards[Review monitoring dashboards]
+
+
+As a cluster administrator you xref:../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues[can determine why]:
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/index.adoc:21]: re-phrased

File: monitoring/index.adoc [hunk 11]
```diff
@@ -0,0 +1,24 @@
+[id="monitoring-overview"]
+= Monitoring overview
+include::modules/common-attributes.adoc[]
+:context: monitoring-overview
+
+toc::[]
+
+The growth of containerization has not only brought many advantages to developers, DevOps, and IT teams but also challenges. The flexibility and scalability of containerized applications pose challenges since there is no direct relationship between an application and the server it runs on. A cluster administrator can use the pre-installed, pre-configured, and self updating {product-title} xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[monitoring stack] to monitor:
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/index.adoc:8]: since I am adding relevant content to the existing `Understanding the monitoring stack` , wont be adding this content

**Comment by bburt-rh** [comment]
- In the preview, I'm seeing two identical "Understanding the monitoring stack" headings in a row, but each has different content. The first section can remain as-is, I think, as a general overview that defines Monitoring. The second heading should be renamed to something like "About monitoring stack components" or whatever you think would be best because this section provides more detail about the system components that make up the stack.  Then the next section lists the components in a table.
- Other than this, it lgtm.

**Comment by Amrita42** [comment]
Good point @bburt-rh  let me go ahead and make the changes .

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 12]
```diff
@@ -9,18 +9,26 @@ toc::[]
...
+
+A cluster administrator can xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations. You also have the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
+
 include::modules/monitoring-targets-for-user-defined-projects.adoc[leveloffset=+2]
 
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics, alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI. You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to perform the following tasks:
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
+* xref:../monitoring/reviewing-monitoring-dashboards.adoc#reviewing-monitoring-dashboards[Review monitoring dashboards]
```
Comment (ctauchen) [inline monitoring/understanding-the-monitoring-stack.adoc:25]: This link is already present in the doc. Line 286. Is there an advantage to having it twice?

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 13]
```diff
@@ -9,18 +9,26 @@ toc::[]
...
 include::modules/monitoring-components-for-monitoring-user-defined-projects.adoc[leveloffset=+2]
+
+A cluster administrator can xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations. You also have the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
+
 include::modules/monitoring-targets-for-user-defined-projects.adoc[leveloffset=+2]
 
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics, alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI. You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to perform the following tasks:
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
```
Comment (ctauchen) [inline monitoring/understanding-the-monitoring-stack.adoc:24]: @pneedle-rh These links weren't present in your original version. Are they new? Any thoughts on how they could be most easily integrated?

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 14]
```diff
@@ -9,18 +9,26 @@ toc::[]
...
 include::modules/monitoring-default-monitoring-components.adoc[leveloffset=+2]
 include::modules/monitoring-default-monitoring-targets.adoc[leveloffset=+2]
 include::modules/monitoring-components-for-monitoring-user-defined-projects.adoc[leveloffset=+2]
+
+A cluster administrator can xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations. You also have the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
+
 include::modules/monitoring-targets-for-user-defined-projects.adoc[leveloffset=+2]
 
+The {product-title} web console provides xref:../monitoring/accessing-third-party-uis.adoc#accessing-third-party-uis[third-party integrated UIs] such as metrics, alerts, and review of monitoring dashboards. You can access these third-party user interfaces (UIs) either through the web console or the CLI. You can use the **Administrator** or ** Developer** perspective in the {product-title} monitoring dashboard to perform the following tasks:
```
Comment (ctauchen) [inline monitoring/understanding-the-monitoring-stack.adoc:21]: I get a bit confused here. Is this section supposed to be part of modules/monitoring-targets-for-user-defined-projects.adoc? Because it appears that way in the HTML, under the same title. This text appears to be mostly about the console, not user-defined targets.

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 15]
```diff
@@ -9,18 +9,26 @@ toc::[]
...
+
+* xref:../monitoring/managing-metrics.adoc#managing-metrics[Manage metrics]
+* xref:../monitoring/managing-alerts.adoc#managing-alerts[Manage alerts]
+* xref:../monitoring/reviewing-monitoring-dashboards.adoc#reviewing-monitoring-dashboards[Review monitoring dashboards]
+
+
+As a cluster administrator, you can find answers to common problems such as user metrics unavailability and Prometheus consuming a lot of disk space in xref:../monitoring/troubleshooting-monitoring-issues.adoc#troubleshooting-monitoring-issues[troubleshooting monitoring issues].
```
Comment (ctauchen) [inline monitoring/understanding-the-monitoring-stack.adoc:28]: Again, this seems not to belong to the section. Would this work as a link under Additional Resources?

File: welcome/learn_more_about_openshift.adoc [hunk 16]
```diff
@@ -54,7 +54,7 @@ Use the following sections to find content to help you learn about and use {prod
 
 | link:https://learn.openshift.com/?extIdCarryOver=true&sc_cid=701f2000001Css5AAC[OpenShift Interactive Learning Portal]
 | xref:../networking/understanding-networking.adoc#understanding-networking[Networking]
-| xref:../monitoring/understanding-the-monitoring-stack.#understanding-the-monitoring-stack[Monitoring]
+| xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[Monitoring]
```
Comment (ctauchen) [inline welcome/learn_more_about_openshift.adoc:57]: Good catch!

**Review by ctauchen — COMMENTED** [review]
Thanks for this update. It's a tricky one, for sure. I've made a few comments about my initial impressions. Please let me know what you think.

I'm also asking @pneedle-rh to take a look, since he knows the subject and the article much better than I do. 

Thanks!

File: modules/about-monitoring-stack-components.adoc [hunk 17]
```diff
@@ -7,8 +7,8 @@
 // can be re-used in associated products but the title is not
 // included in the existing OpenShift assembly.
 
-[id="understanding-the-monitoring-stack_{context}"]
-= Understanding the monitoring stack
+[id="about-monitoring-stack-components_{context}"]
+= About monitoring stack components
```
[the remark this answers is not in the document]
Comment (pneedle-rh) [reply modules/about-monitoring-stack-components.adoc:11]: @Amrita42 thank you for building on this page in the Monitoring book!

This section is about more than the components in the monitoring stack. If the reason for the heading change for the section is primarily to remove the duplication with the main assembly title, you could instead change the main title and leave the section heading as it is.

For example, the main assembly title could be something like "Overview of monitoring" and the heading for this section could remain as "Understanding the monitoring stack".

The title would then be consistent with the overview pages in the other books and the heading for this section would continue to describe the content holistically.

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 18]
```diff
@@ -9,18 +9,26 @@ toc::[]
...
+
 == Additional resources
 
 * xref:../support/remote_health_monitoring/about-remote-health-monitoring.adoc#about-remote-health-monitoring[About remote health monitoring]
 * xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#granting-users-permission-to-monitor-user-defined-projects_enabling-monitoring-for-user-defined-projects[Granting users permission to monitor user-defined projects]
 * xref:../security/tls-security-profiles.adoc#tls-security-profiles[Configuring TLS security profiles]
-
-== Next steps
```
Comment (pneedle-rh) [inline monitoring/understanding-the-monitoring-stack.adoc:24]: When the Monitoring book was restructured for 4.6, at the introduction of monitoring for user-defined projects, the assemblies were developed to build on each other sequentially. After reading about the concepts relating to the monitoring stack on this page, the intention was for the next step in the reader's journey to be "Configuring the monitoring stack".

The steps in the "Configuring the monitoring stack" page must be reviewed and configured according to the customer's needs before the procedures in the remaining assemblies. My view is that the "Next steps" section provides a clear direction as to what should be done next, then the "Additional resources" section provides further reading on topics related to the material in this page. "Next steps" and "Additional resources" sections are consistent with other pages in this book and in the library.

What is the underlying requirement behind removing this or renaming it?

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 14 above)
Comment (pneedle-rh) [reply monitoring/understanding-the-monitoring-stack.adoc:21]: I can see the benefit to the reader to adding this planning content to summarise the metrics, alerts and dashboard capabilities, plus the troubleshooting monitoring topic, with links to the associated pages.

You could create a new section for this, outside of the context of monitoring targets for user-defined projects, because that is a separate topic.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 13 above)
Comment (pneedle-rh) [reply monitoring/understanding-the-monitoring-stack.adoc:24]: I can see the value in having a planning section that provides a summary of these features and then provides links to those pages. I think this should be in a new section, as mentioned in my preceding comment just now.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 12 above)
Comment (pneedle-rh) [reply monitoring/understanding-the-monitoring-stack.adoc:25]: You could have a dedicated section that describes he metrics, alerts and dashboard capabilities, plus the troubleshooting monitoring topic and provides links, then determine if you need an additional resources section in addition to that.

File: welcome/index.adoc [hunk 19]
```diff
@@ -281,7 +281,7 @@ xref:../updating/updating-disconnected-cluster.adoc#updating-disconnected-cluste
 
 - **xref:../logging/cluster-logging.adoc#cluster-logging[Work with OpenShift Logging]**: Learn about OpenShift Logging and configure different OpenShift Logging types, such as Elasticsearch, Fluentd, and Kibana.
 
-- **xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[Monitor clusters]**:
+- **xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[monitor clusters]**:
```
Comment (pneedle-rh) [inline welcome/index.adoc:284]: You could keep "Monitor" as upper case, given that it is the first word in the bulleted item.

File: welcome/learn_more_about_openshift.adoc (hunk 16 above)
Comment (pneedle-rh) [reply welcome/learn_more_about_openshift.adoc:57]: +1

File: welcome/learn_more_about_openshift.adoc [hunk 20]
```diff
@@ -98,7 +98,7 @@ Use the following sections to find content to help you learn about and use {prod
 |
 
 |
-| xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[Monitoring]
+| xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[Monitoring stack]
```
Comment (pneedle-rh) [inline welcome/learn_more_about_openshift.adoc:101]: Looking at the other items in this column, "Monitoring" seems consistent and it is more than just the monitoring stack.

**Review by pneedle-rh — COMMENTED** [review]
@Amrita42 thank you for developing this page in the Monitoring book. I have now completed my review. Please let me know if you have any questions about my suggestions. Thanks.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 12 above)
Comment (pneedle-rh) [reply monitoring/understanding-the-monitoring-stack.adoc:25]: Thinking about it further, perhaps the new section could also include a description and link to all of the other pages in the Monitoring book, in addition to the ones that are listed here. If the requirement is for a top level overview of the content in the book and links to relevant pages, then it would be useful for that to be a complete list. Perhaps the section could be called something like "Preparing to use the monitoring stack" or something similar.

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 21]
```diff
@@ -9,18 +9,26 @@ toc::[]
 include::modules/monitoring-about-cluster-monitoring.adoc[leveloffset=+1]
 
 // Understanding the monitoring stack
-include::modules/monitoring-understanding-the-monitoring-stack.adoc[leveloffset=+1]
+include::modules/about-monitoring-stack-components.adoc[leveloffset=+1]
 include::modules/monitoring-default-monitoring-components.adoc[leveloffset=+2]
 include::modules/monitoring-default-monitoring-targets.adoc[leveloffset=+2]
 include::modules/monitoring-components-for-monitoring-user-defined-projects.adoc[leveloffset=+2]
+
+A cluster administrator can xref:../monitoring/configuring-the-monitoring-stack.adoc#configuring-the-monitoring-stack[configure the monitoring stack] with the supported configurations. You also have the option to enable xref:../monitoring/enabling-monitoring-for-user-defined-projects.adoc#enabling-monitoring-for-user-defined-projects[monitoring for user-defined projects].
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:17]: This is to give an overview of the entire Monitoring Book and more appropriate here then Additional Resources imo.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 12 above)
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:25]: I agree with that @pneedle-rh , objective is to give the user an overall view of whats there in the entire book. If those links are present then we dont  need a repetition of Additional Resources.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 13 above)
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:24]: I agree

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 14 above)
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:21]: done

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 15 above)
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:28]: Created a section where this content is added.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 18 above)
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:24]: Yeah given the context , I have added back the next steps.

File: monitoring/understanding-the-monitoring-stack.adoc (hunk 21 above)
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:17]: I have retained two links that link to different books for Additional Resources.

**Comment by Amrita42** [comment]
@pneedle-rh Thanks for the extensive review. Could you take a look at the new preview?

File: monitoring/overview-of-monitoring.adoc [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+
+// Understanding the monitoring stack
+include::modules/understanding-the-monitoring-stack.adoc[leveloffset=+1]
+include::modules/monitoring-default-monitoring-components.adoc[leveloffset=+2]
+include::modules/monitoring-default-monitoring-targets.adoc[leveloffset=+2]
+include::modules/monitoring-components-for-monitoring-user-defined-projects.adoc[leveloffset=+2]
+include::modules/monitoring-targets-for-user-defined-projects.adoc[leveloffset=+2]
+
+[id="how-to-use-the-monitoring-stack"]
+= How to use the monitoring stack?
```
[the remark this answers is not in the document]
Comment (pneedle-rh) [reply monitoring/overview-of-monitoring.adoc:19]: My feeling is that this new section duplicates a lot of the content that is included in the "About OpenShift Container Platform monitoring" section at the start of the assembly.

If the intention is to include links to the various resources included within the Monitoring book, you could move the existing "About OpenShift Container Platform monitoring" section from `modules/monitoring-about-cluster-monitoring.adoc` to this `monitoring/overview-of-monitoring.adoc` assembly and add the cross references to the existing content.

My view is that it makes sense also to keep that section at the start of the assembly as it provides a useful introduction for the reader.

**Review by pneedle-rh — COMMENTED** [review]
@Amrita42 thank you for your further additions to the PR. I have now completed my additional review. Please let me know if you have any questions. Thanks.

File: logging/cluster-logging-dashboards.adoc [hunk 23]
```diff
@@ -9,11 +9,11 @@ The *Logging/Elasticsearch Nodes* and *Openshift Logging* dashboards in the {pro
 
 The *OpenShift Logging* dashboard contains charts that show details about your Elasticsearch instance at a cluster level, including cluster resources, garbage collection, shards in the cluster, and Fluentd statistics.
 
-The *Logging/Elasticsearch Nodes* dashboard contains charts that show details about your Elasticsearch instance, many at node level, including details on indexing, shards, resources, and so forth. 
+The *Logging/Elasticsearch Nodes* dashboard contains charts that show details about your Elasticsearch instance, many at node level, including details on indexing, shards, resources, and so forth.
 
 [NOTE]
 ====
-For more detailed data, click the *Grafana UI* link in a dashboard to launch the Grafana dashboard. Grafana is shipped with xref:../monitoring/understanding-the-monitoring-stack.adoc#understanding-the-monitoring-stack[OpenShift cluster monitoring].
+For more detailed data, click the *Grafana UI* link in a dashboard to launch the Grafana dashboard. Grafana is shipped with xref:../monitoring/overview-of-monitoring.adoc#overview-of-monitoring[Overview of monitoring].
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply logging/cluster-logging-dashboards.adoc:16]: Re-phrased as : Grafana is shipped with xref:../monitoring/overview-of-monitoring.adoc#overview-of-monitoring [OpenShift cluster monitoring] since the title is now `overview of monitoring`

File: modules/about-monitoring-stack-components.adoc [hunk 24]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * virt/logging_events_monitoring/virt-openshift-cluster-monitoring.adoc
+// * monitoring/understanding-the-monitoring-stack.adoc
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply modules/about-monitoring-stack-components.adoc:4]: Good point!

[the remark this answers is not in the document]
Comment (Amrita42) [reply modules/about-monitoring-stack-components.adoc:4]: Done for the six modules.

File: monitoring/overview-of-monitoring.adoc (hunk 22 above)
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/overview-of-monitoring.adoc:19]: @pneedle-rh Are you saying remove the `include::modules/monitoring-about-cluster-monitoring.adoc[leveloffset=+1]` at the start of the assembly and add it as an xref in the new section `How to use the monitoring stack?`. Yes the objective is to give an overall view/relevant links to the entire book.

File: monitoring/understanding-the-monitoring-stack.adoc [hunk 25]
```diff
@@ -9,18 +9,26 @@ toc::[]
 include::modules/monitoring-about-cluster-monitoring.adoc[leveloffset=+1]
 
 // Understanding the monitoring stack
-include::modules/monitoring-understanding-the-monitoring-stack.adoc[leveloffset=+1]
+include::modules/about-monitoring-stack-components.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/understanding-the-monitoring-stack.adoc:12]: Correct , this assembly is no longer valid as its duplicated as Overview of monitoring. Looks like I didnt remove this file . I will use Git mv in future but now I have already duplicated the file.

File: monitoring/overview-of-monitoring.adoc [hunk 26]
```diff
@@ -0,0 +1,37 @@
+[id="overview-of-monitoring"]
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/overview-of-monitoring.adoc:1]: Noted

File: modules/about-monitoring-stack-components.adoc [hunk 27]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
```
[the remark this answers is not in the document]
Comment (Amrita42) [reply modules/about-monitoring-stack-components.adoc:1]: The module is not required as I have used `modules/understanding-the-monitoring-stack.adoc`. Removing the module.

File: monitoring/overview-of-monitoring.adoc (hunk 22 above)
[the remark this answers is not in the document]
Comment (pneedle-rh) [reply monitoring/overview-of-monitoring.adoc:19]: I was suggesting that you could move the entire contents of `modules/monitoring-about-cluster-monitoring.adoc` from that module file into the `modules/about-monitoring-stack-components.adoc` assembly itself. With the contents being in the assembly instead of the module, you could then add cross-references within the existing content, to the different pages in the Monitoring book. This would remove the need for an additional section such as the "How to use the monitoring stack?" section, which duplicates a lot of the contents that are in "About OpenShift Container Platform monitoring".

In other words, instead of adding the "How to use the monitoring stack?" section into the assembly, you could have the "About OpenShift Container Platform monitoring" section in the assembly (not in a module) and include cross-references to other pages in that existing content.

[the remark this answers is not in the document]
Comment (pneedle-rh) [reply monitoring/overview-of-monitoring.adoc:19]: It's just an idea for your consideration, to reduce duplication in the page.

[the remark this answers is not in the document]
Comment (pneedle-rh) [reply monitoring/overview-of-monitoring.adoc:19]: @ctauchen what are your thoughts on this? Thanks!

[the remark this answers is not in the document]
Comment (pneedle-rh) [reply monitoring/overview-of-monitoring.adoc:19]: Looking at the draft as it stands now, I noticed the duplication between the sections. This wasn't something I had considered previously and I think it is worth trying to minimise that. My apologies if my further suggestions have created additional work for you, that was not my intention.

[the remark this answers is not in the document]
Comment (Amrita42) [reply monitoring/overview-of-monitoring.adoc:19]: On further thought, I like your suggestion of adding the content from the module` modules/monitoring-about-cluster-monitoring.adoc` and then adding what's relevant from the new section `[id="how-to-use-the-monitoring-stack"]`.

[the remark this answers is not in the document]
Comment (ctauchen) [reply monitoring/overview-of-monitoring.adoc:19]: > @ctauchen what are your thoughts on this? Thanks!

I think that's a good solution. Incorporate the new content cleanly into the existing structure. It's slightly messy to put it all in the assembly. With tools limitations, that's probably the best option.

Nearly there!

File: modules/about-monitoring-stack-components.adoc (hunk 27 above)
Comment (ctauchen) [inline modules/about-monitoring-stack-components.adoc:1]: This file isn't included in the assembly.

File: _topic_maps/_topic_map.yml [hunk 28]
```diff
@@ -1998,8 +1998,8 @@ Name: Monitoring
 Dir: monitoring
 Distros: openshift-enterprise,openshift-origin
 Topics:
-- Name: Understanding the monitoring stack
-  File: understanding-the-monitoring-stack
+- Name: Overview of monitoring
```
Comment (ctauchen) [inline _topic_maps/_topic_map.yml:2001]: I hate to bring this up, given the amount of changes. But we have many books that use the pattern **BlaBlaBla overview**, rather than **Overview of BlaBlaBla**. See Architecture, Installing, Support, and so on.

Whatever the merits of one construction over another, we should always aim for consistency wherever possible.

File: modules/about-monitoring-stack-components.adoc (hunk 27 above)
Comment (Amrita42) [reply modules/about-monitoring-stack-components.adoc:1]: The file has been re-named .  Content is the same. Based on previous feedback.

**Comment by Amrita42** [comment]
Update>> Will have to close this PR and link the PR due to conflicts.

**Comment by Amrita42** [comment]
Closing this PR refer https://github.com/openshift/openshift-docs/pull/41356
